subsys/bluetooth/host/conn.c

```diff
@@ -750,8 +750,14 @@ static void conn_destroy(struct bt_conn *conn, void *data)
 	k_work_cancel_delayable(&conn->deferred_work);
 
 	conn->pending_no_cb = 0;
+	conn->tx_pending.head = NULL;
+	conn->tx_pending.tail = NULL;
```

Comment:
Alternatively we can call `sys_slist_init(conn->tx_pending)` which I guess does the same, but using the `slist` API